pkg/kademlia/kademlia.go

```diff
@@ -322,7 +322,9 @@ func (k *Kademlia) refresh(ctx context.Context, threshold time.Duration) error {
 		if tErr != nil {
 			errors.Add(tErr)
 		} else if now.After(ts.Add(threshold)) {
-			rID, _ := randomIDInRange(startID, keyToBucketID(bID))
+			endID := keyToBucketID(bID)
+			rID, _ := randomIDInRange(startID, endID)
+			startID = endID
```

Comment:
this doesn't look right

so if we have buckets:

1. refresh
2. not stale
3. not stale
4. needs refresh

then the last refresh would refresh buckets 2-4, which might miss 4.

Reply:
doh!  fixed it